src/models/hooks/activityReport.js

```diff
@@ -45,16 +45,23 @@ const propogateSubmissionStatus = async (sequelize, instance, options) => {
         includeIgnoreAttributes: false,
         transaction: options.transaction,
       });
-      const templateIds = await Promise.all(goals.map(async (goal) => findOrCreateGoalTemplate(
-        sequelize,
-        options.transaction,
-        instance.regionId,
-        goal.name,
-        goal.createdAt,
-        goal.updatedAt,
-      )));
-      await Promise.all(goals.map(async (goal, i) => sequelize.models.Goal.update(
-        { goalTemplateId: templateIds[i] },
+      const distinctGoals = [...new Map(goals.map((goal) => [goal.name, goal])).values()];
```

Comment:
Would love to see some comments here, even like "create a distinct list of goals based on name".

I'm also wondering if "propogateSubmissionStatus" is kind of poor name for this function or at the least, if the function is getting kind of overloaded